src/ripple/basics/Slice.h

```diff
@@ -131,9 +131,13 @@ inline
 bool
 operator== (Slice const& lhs, Slice const& rhs) noexcept
 {
-    return lhs.size() == rhs.size() &&
-        std::memcmp(
-            lhs.data(), rhs.data(), lhs.size()) == 0;
+    if (lhs.size() != rhs.size())
+        return false;
+
+    if (lhs.size() == 0)
+        return true;
+
+    return std::memcmp(lhs.data(), rhs.data(), lhs.size()) == 0;
```

Comment:
```
return lhs.size() == rhs.size() &&
  (lhs.size() == 0 ||
    std::memcmp(lhs.data(), rhs.data(), lhs.size()) == 0);
```
Also, I see this is covered (https://github.com/nbougalis/rippled/blob/65d7f91435bb710f3eac4e88986affc03c025229/src/ripple/basics/Slice.h#L132), but could it use some explicit tests?